sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -364,11 +364,24 @@ class Analyzer(
 
         val (resolvedOrdering, missing) = resolveAndFindMissing(ordering, a, groupingRelation)
 
-        if (missing.nonEmpty) {
+        val addForAlias = new ArrayBuffer[NamedExpression]()
+        val aliasedOrdering = resolvedOrdering.zipWithIndex.map {
+          case (o, i) => {
+            o transform {
+              case aggOrSub @ (_: AggregateExpression | _: Substring) =>
```

Comment:
@cloud-fan Because `Substring` has the same problem as `Aggregation`.
I think we need to add another rule to solve your 2st commit, because there're many sqls like `select key, sum(value), sum(value) from src group by key` have the same problem.